packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -139,9 +139,12 @@ export class S3Provider implements StorageProviderInterface {
         profile: config.aws.s3.roleArn ? 'role' : 'default',
         filepath: credentialsPath
       }),
-      endpoint: config.server.storageProviderExternalEndpoint
-        ? config.server.storageProviderExternalEndpoint
-        : config.aws.s3.endpoint,
+      endpoint:
+        config.server.storageProviderExternalEndpoint?.length > 0
```

Comment:
can we use `typeof` instead of `.length` ?

Reply:
oh, it defaults to an empty string right. we can just use an equality check, the `?.length > 0 is redundant`

Reply:
Check that it's `!== ''`, or something else?

Reply:
Nevermind, I think I see what you're getting at, we can do externalEndpoint || s3.endpoint || undefined and it'll treat empty strings or null/undefined as falsey and then default to undefined.